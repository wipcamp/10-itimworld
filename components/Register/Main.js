import React from 'react'
import { compose } from 'recompose'
import { reduxForm } from 'redux-form'
import { connect } from 'react-redux'
import styled from 'styled-components'
import {actions as registerActions} from '../../store/reducers/register'

import { fields } from './form.json'
import FieldInput from '../Core/Input'

const SubHeader = styled.h3`
  background: #336699;
  color: #fff;
  padding: 10px 0;
  margin: 0;
  border-radius: 10px 10px 0 0;
`

const BackgroundContainer = styled.div`
  background: #252525;
  min-height: 100vh;
  height: auto;
  background-repeat: no-repeat;
  background-size: cover;
  background-attachment: fixed;
`

const RegisterSection = styled.form`
  color: #032e51;
  background: #fff;
  margin-bottom: 5em;
  border-radius: 10px;
`

const SubmitButton = styled.button`
  background-color: #336699;
  color: #fff;
  font-weight: bold;

  &:hover {
    cursor: ${props => props.disabled ? 'no-drop' : 'pointer'} ;
  }
`

const FormRegister = props => {
  const { handleSubmit, pristine, submitting, saveRegister } = props
  console.log('show vat > ', props)
  return (
    <RegisterSection onSubmit={handleSubmit(saveRegister)}>
      <SubHeader>ลงทะเบียน</SubHeader>
      <div className='row px-3 pb-5 pt-3'>
        {
          fields.map((field, index) => (
            <FieldInput key={index} {...field} />
          ))
        }
        <div className='col-12'>
          <SubmitButton
            className={'btn btn-lg'}
            type='submit'
            disabled={pristine || submitting}
            title={'ถัดไป'}
          >
          ถัดไป
          </SubmitButton>
        </div>
      </div>
    </RegisterSection>
  )
}

export const MainRegister = props => (
  <div>
    <BackgroundContainer>
      <div className='container'>
        <div className='row '>
          <div className='col-12 col-md-6 mx-auto text-center justify-content-center'>
            <img src='https://wip.camp/assets/img/logo/wipcamp9-full-transparent.svg' className='img-fluid' alt='wipcamp-logo' />
          </div>
          <div className='col-12 text-center text-white'>
            <div>
              ขั้นตอนการลงทะเบียน
            </div>
          </div>
          <div className='col-12 col-sm-10 mx-auto text-center' >
            <FormRegister {...props} />
          </div>
        </div>
      </div>
    </BackgroundContainer>
  </div>
)

export default compose(
  connect(
    state => ({
      temp: state
    }),
    {...registerActions}
  ),
  reduxForm({
    form: 'register',
    initialValues: {
      user_id: 10
    }
  })
)(MainRegister)
