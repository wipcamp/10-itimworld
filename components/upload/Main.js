import React from 'react'
import styled from 'styled-components'

import Newheader from './header'

const BackgroundContainer = styled.div`
  background: #B8D0EC;
  min-height: 100vh;
  height: auto;
`

const CardUpload = styled.div`
  font-family: 'pridi-regularr';
  font-size: 28px;
  color: #B8D0EC;

  & > input[type=file] {
    position: absolute;
    z-index: -1;
    width: 10px;
    
    &:focus + label {
      text-shadow: 0 0 10px rgba(81,255,255,1);
      
    }
  }

  & > label {
    margin: 20px auto;
    display: flex;
    justify-content: center;
    width: 290px;
    height: 230px;
    cursor: pointer;

    align-items: center;
    background-size: cover;
    background-image: url(${props => props.img});

    @media only screen and (min-width: 768px) and (max-width: 991px) {
      width: 210px;
      height: 168px;
    }

    @media(min-width: 992px) {
      margin-top: 100px;
      margin: 100px ${props => props.margin};
    }
  }
`

const Card = ({ img, name, margin, content, outerClass }) => (
  <div className={outerClass}>
    <CardUpload img={img} margin={margin}>
      <input type='file' id={`${name}-file-input`} />
      <label
        htmlFor={`${name}-file-input`}
        dangerouslySetInnerHTML={{ __html: content }}
      />
    </CardUpload>
  </div>
)

const cardData = [
  {
    name: '1',
    outerClass: 'col-12 col-md-4 pr-md-0',
    margin: '0 0 auto',
    img: '/static/img/upload-card-1.png',
    content: 'ตอบคำถาม'
  },
  {
    name: '2',
    outerClass: 'col-12 col-md-4 px-md-0',
    margin: 'auto 0' ,
    img: '/static/img/upload-card-2.png',
    content: 'อัพโหลดใบ ปพ.1'
  },
  {
    name: '3',
    outerClass: 'col-12 col-md-4 pl-md-0',
    margin: 'auto 0 0',
    img: '/static/img/upload-card-1.png',
    content: 'อัพโหลดใบเอกสาร<br />ขออนุญาตผู้ปกครอง'
  }
]

const MainUpload = props => (
  <div>
    <BackgroundContainer>
      <Newheader img={`https://pbs.twimg.com/profile_images/829362291237801985/mvlVSd7J.jpg`} />
      <UploadList {...props} />
    </BackgroundContainer>
  </div>
)

const UploadList = () => (
  <div className='container'>
    <div className='row text-center'>
      {
        cardData.map((data, index) => (
          <Card
            key={index}
            name={data.name}
            margin={data.margin}
            img={data.img}
            outerClass={data.outerClass}
            content={data.content}
          />
        ))
      }
    </div>
  </div>
)

export default MainUpload
