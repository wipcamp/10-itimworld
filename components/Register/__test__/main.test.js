/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'

import { MainRegister } from '../Main'

describe('<MainRegister />', () => {
  it('MainRegister should be renders', () => {
    const mockupProps = {
      firstName: ''
    }
    const component = renderer.create(<MainRegister register={mockupProps} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
