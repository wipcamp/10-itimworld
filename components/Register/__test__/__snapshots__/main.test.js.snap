// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainRegister /> MainRegister should be renders 1`] = `
<div>
  register component
  <h1>
    
  </h1>
  <div>
    <label
      htmlFor="input-firstName"
    >
      firstname
      :
    </label>
    <input
      className="form-control"
      id="input-firstName"
      onChange={[Function]}
      placeholder="Input your name"
      required={true}
      type="text"
      value=""
    />
  </div>
</div>
`;
